astroquery/dace/core.py

```diff
@@ -0,0 +1,75 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import print_function
+from collections import defaultdict
+from astropy.table import Table
+from ..query import BaseQuery
+from ..utils import async_to_sync
+from . import conf
+
+__all__ = ['Dace', 'DaceClass']
+
+
+class ParseException(Exception):
+    """Raised when parsing Dace data fails"""
+
+
+@async_to_sync
+class DaceClass(BaseQuery):
+    """
+    DACE class to access DACE (Data Analysis Center for Exoplanets) data based in Geneva Observatory
+    """
+
+    __DACE_URL = conf.server
+    __DACE_TIMEOUT = conf.timeout
+    __OBSERVATION_ENDPOINT = 'ObsAPI/observation/'
+    __RADIAL_VELOCITIES_ENDPOINT = __OBSERVATION_ENDPOINT + 'radialVelocities/'
+    __HEADERS = {'Accept': 'application/json'}
+
+    def query_radial_velocities_async(self, object_name):
```

Comment:
Thanks for all your comments ! I will update the code.
About query_object, I think I will implement it in a second step. query_radial_velocities_async is named like this because it really return only radial velocities data.
For query_object we have another API on our side to call but it is not fully implemented.
I created only radial_velocities method for now because this is a request from our astrophysicists.
To summarize, if it is ok for you I'll start with this and will add query_object later.

Reply:
And what would `query_object` return in addition to the RV data? I'm also interested in the differences in the data structure of the returns.

Reply:
We rediscussed with astrophysicists on our side to understand their needs regarding the DACE APIs.
At the moment we have two kind of data to retrieve for a given object : 

### **1. time series** 
This is final data product of observations. It could span over hours or years. These are mainly used to search for and characterize time dependent astronomical phenomena such as binary stars, exoplanets companions. It could also be used to study stellar activity on different time scales. 
Such series could be related to :
**High resolution spectroscopic data analysis** 
- radial velocities+ errors
- stellar activity indexes+ errors
- other time dependent parameters linked  to high resolution spectroscopic data analysis
 
**Imaging data analysis**
- Light curves + errors

### **2. Reduced data products.**
Possibility to download reduced data products in the "fits" file format for different instruments we have builded ie.   CORALIE, HARPS, ESPRESSO and soon NIRPS

Based on these requirements,  we suggest to implement the following methods :
1. **retrieve_time_series**(object_name, type=spectroscopy/photometry/etc., columns=[list of columns]). For « type » parameter, in this first pull request only spectroscopy will be implemented => rajouter des options. rv, activity index etc...
2. **retrieve_high_resolution_spectroscopy_data**(fits_file_list). A list of fits file to download

Best regards

Reply:
Don't hesitate if you have questions, suggestions, etc.
I tried to explain the needs from researchers, hope it will help to take a decision

Reply:
About query_object, initially my idea was to return also a Table astropy with stellar data : ra, dec, magv, etc. but it is not implemented on our side at the moment.

Reply:
Any news ?

Reply:
Hello,

Don't hesitate to come back to me. I close the discussion for now. i let you decide what's best and I'll implement it.

Regards,

Julien